v1.3: Experimental features API route (#2496)
meilisearch/documentation: reference/api/experimental-features.mdx

# Experimental

> The `/experimental-features` route configures Meilisearch's [experimental features](/learn/experimental/overview). All operations in this route are synchronous.

Reviewer: ```suggestion
The `/experimental-features` route allows you to activate or deactivate some of Meilisearch's [experimental features](/learn/experimental/overview). All operations in this route are synchronous.
```

This could just be a personal feeling, but I would want to clarify the second sentence. Example:

```suggestion
The `/experimental-features` route allows you to activate or deactivate some of Meilisearch's [experimental features](/learn/experimental/overview).

This route is **synchronous**. This means that no task object will be returned, and any activated or deactivated features will be made available or unavailable immediately.
```

<Capsule intent="danger">

> The experimental API route is not compatible with all experimental features. Consult the [experimental feature overview](/learn/experimental/overview) for a compatibility list.

Reviewer: ```suggestion
The experimental API route is not compatible with all experimental features. Consult the [experimental feature overview](/learn/experimental/overview#current-experimental-features) for a compatibility list.
```

</Capsule>

## Experimental features object

```json
{
  "scoreDetails": false,
  "vectorSearch": false
}
```

| Name               | Type    | Description                                              |
|-:------------------|-:-------|-:--------------------------------------------------------|

> | **`scoreDetails`** | Boolean | `true` if feature is currently active, `false` otherwise |

Reviewer: ```suggestion
| **`scoreDetails`** | Boolean | `true` if feature is active, `false` otherwise |
```

> | **`vectorSearch`** | Boolean | `true` if feature is currently active, `false` otherwise |

Reviewer: ```suggestion
| **`vectorSearch`** | Boolean | `true` if feature is active, `false` otherwise |
```

## Get all experimental features

<RouteHighlighter method="GET" route="/experimental-features"/>

> Get a list of all experimental features that can be configured during runtime and their current status.

Reviewer: ```suggestion
Get a list of all experimental features that can be activated via the `/experimental-features` route and whether or not they are currently activated.
```
